pkg/util/validation/limits.go

```diff
@@ -46,6 +46,12 @@ type DisabledRuleGroup struct {
 
 type DisabledRuleGroups []DisabledRuleGroup
 
+type HighPriorityQuery struct {
+	Regex     string        `yaml:"regex" doc:"nocli|description=Query string regex. If evaluated true (on top of meeting all other criteria), query is treated as a high priority."`
```

Comment:
Now we do `regexp.MatchString` to match regex. This is expensive because we need to compile regex every time when we match.

Can we also try to store the complied version of regex here?